velox/exec/PartitionedOutputBufferManager.h

```diff
@@ -40,22 +40,62 @@ struct DataAvailable {
   }
 };
 
-class DestinationBuffer {
+/// The class is used to buffer the output pages which haven't been fetched by
+/// any destination for arbitrary output.
+///
+/// NOTE: this class is not thread-safe.
+class ArbitraryBuffer {
```

Comment:
Is there always zero or one arbitrary buffer? Can there be multiple such buffers?

Reply:
There is zero for non-arbitrary output type and only one for arbitrary output. Commented to make it clear.